### ISSUE TITLE
Discovery example breaks on resize on desktop

### ISSUE THREAD

**Issue by TimWhiting** [fzyzcjy/flutter_portal#43]
**Describe the bug**
In certain conditions a full-size `PortalEntry`'s size is not updated when resizing the window.

**To Reproduce**

- Replace the discovery example's build method with this. I can't get it to reproduce with the `PortalEntry` around the floating action button, but around the other button it reproduces consistently.
```dart
 @override
  Widget build(BuildContext context) {
    return Portal(
      child: MaterialApp(
        home: Scaffold(
          appBar: AppBar(title: const Text('Discovery example')),
          body: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                const Text('You have clicked the button this many times:'),
                Text('$count', style: Theme.of(context).textTheme.headline4),
                Discovery(
                    visible: showDiscovery,
                    description: const Text('Click to increment the counter'),
                    onClose: () => setState(() => showDiscovery = false),
                    child: ElevatedButton(
                      onPressed: () => setState(() => showDiscovery = true),
                      child: const Text('Show discovery'),
                    )),
              ],
            ),
          ),
          floatingActionButton: FloatingActionButton(
            onPressed: _increment,
            child: const Icon(Icons.add),
          ),
        ),
      ),
    );
  }
```
- Generate the desktop files by running `flutter create .` in the examples folder with a desktop configuration enabled
- Run the app
- Resize the window, it's not super consistent, but it does happen more if you drag the bottom right corner of the window to the far right of the screen and then drag down a ways.

See the following:
<img width="871" alt="Screen Shot 2022-02-17 at 11 15 22 PM" src="https://user-images.githubusercontent.com/15617545/154628491-97ebb5d7-4d2e-49ee-84fb-535663ae9733.png">

**Expected behavior**

The full-size `PortalEntry` overlay should resize like the rest of the window (which does update appropriately), the `PortalEntry` that follows the button is fine, just the barrier doesn't fill the space.

**Comment by fzyzcjy** [comment]
Hi, this issue maybe related, since it is also about problems when resizing: https://github.com/rrousselGit/flutter_portal/issues/53

**Comment by TimWhiting** [comment]
Still is happening, a more reliable way to reproduce the problem is not to resize by the corner, just the bottom of the window. If you drag down quickly you will see it. It doesn't have the same issue for horizontal resizing though.

**Comment by TimWhiting** [comment]
I looked at the issue you mentioned. I think that is the root cause. I switched it to a Row and got horizontal resizing issues instead of vertical. Wrapping in a Flexible or Expanded seems to fix it, I'll close this in favor of #53, but I think the Flexible / Expanded workaround is not great, because we don't always want the widget to fill the full width / height.

**Comment by fzyzcjy** [comment]
Thanks for the information :)
